### PR TITLE
Deb and RPM Installer Support

### PR DESCRIPTION
**Pull request by jnsgruk**
Added support for installing with apt where appropriate.

Deprecated the rc and preprod setup scripts. Similar functionality can be achieved with the `-v` command line switch.

### CODE REVIEW

File: setup.sh [hunk 1]
```diff
@@ -19,48 +13,112 @@ error() { echo -e >&2 "\e[31m[!] ${1:-}\e[0m"; exit 1; }
...
+    sudo apt-get update -qq
+    deps=(apt-transport-https wget)
+    sudo apt-get install -yq "${deps[@]}"
+
+    # Add and trust the Enclave package repository
+    info "Adding Enclave GPG package signing key."
+    wget -qO- https://packages.enclave.io/apt/pubkey.gpg | sudo apt-key add - >/dev/null 2>&1
+    info "Adding the Enclave package repository."
+    # shellcheck disable=SC2024
+    wget -qO- "https://packages.enclave.io/apt/${ENCLAVE_PKG_LIST}" | sudo tee "/etc/apt/sources.list.d/${ENCLAVE_PKG_LIST}" >/dev//null
```
Comment (enclave-alistair) [inline setup.sh]: Is the double-forward slash required when piping `tee` to dev null?

**Review by enclave-alistair — COMMENTED** [review]
This looks great Jon; going to wait until we can get a stable build done before I merge this.

File: setup.sh (hunk 1 above)
Comment (jnsgruk) [reply setup.sh]: Not, that's a typo. One of those weird ones that will actually still work though!

**Comment by jnsgruk** [comment]
Corrected

**Comment by jnsgruk** [comment]
Added an update which should behave properly once the RPM process is integrated. Not tested fully yet because there is no RPM repo, but pretty confident this will get you most of the way there once it's all hosted.

**Comment by enclave-alistair** [comment]
@jnsgruk, when you get a couple of spare minutes, could you update the GPG key URL in the install script from `https://packages.enclave.io/apt/pubkey.gpg` to `https://packages.enclave.io/apt/enclave.stable.gpg`? We just renamed it the other day.

**Pull request closed without merging**